Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -496,17 +496,58 @@ void Painter::draw_ellipse_intersecting(IntRect const& rect, Color color, int th
     if (thickness <= 0)
         return;
 
-    constexpr int number_samples = 100; // FIXME: dynamically work out the number of samples based upon the rect size
-    float increment = AK::Pi<float> / number_samples;
-
-    auto ellipse_xy = [&rect](float theta) -> IntPoint {
-        float s, c;
-        AK::sincos(theta, s, c);
-        return IntPoint { (c * rect.width() * AK::Sqrt1_2<float>), (s * rect.height() * AK::Sqrt1_2<float>)} + rect.center();
+    auto const x_center = rect.center().x();
+    auto const y_center = rect.center().y();
+
+    auto const draw_real_world_x4 = [this, &color, thickness, x_center, y_center](int x, int y) {
+        draw_line({ x_center + x, y_center + y }, { x_center + x, y_center + y }, color, thickness);
```

Comment:
Is all four draw_line calls effectively set one pixel (with the exception of sickness)? I.e. start and end point matches.
In this case, maybe it is better to be explicit:

```cpp
auto const center = rect.center();

auto const draw_real_world_x4 = [this, &color, thickness, center](int x, int y) {
...
const IntPoint directions[4] = { {x, y}, {x, -y}, {-x, y}, {-x, -y}};
for (auto const& delta : directions) {
    auto const point = center.translated(delta);
    draw_line(point, point, color, thickness);
}

};
```

Reply:
Yep, definitely better!